Wordle.Api/Wordle.Api/Wordle.Api/Services/WordService.cs

```diff
@@ -76,5 +77,80 @@ public async Task<string> GetWordOfTheDay(TimeSpan offset, DateTime? date = null
                 }
             }
         }
+
+        public async Task<string[]> GetAllWords() 
+        {
+            var words = await _db.Words.OrderBy(word => word.Text).ToArrayAsync();
+
+            var wordArray = new string[words.Length];
+            var count = 0;
+
+            foreach (var word in words) 
+            {
+                wordArray[count] = $"{words[count].Text}";
+                count++;
+            }
+
+            return wordArray;
+        }
```

Comment:
You do not need the foreach loop at all as the ToListAsync() method is taking care of that for you.
```suggestion
        public async Task<List<Word>> GetAllWords() 
        {
            return await _db.Words.OrderBy(word => word.Text).ToListAsync();
        }
```
You should also use lists instead of arrays. The same thing goes for the GetAllIsCommon() method. 
Either way you're gonna want to change this method to implement pagination to only return chunks of data at a time, then you're front end won't be laggy.

Reply:
Awesome, thank you!